### PR TITLE
#24514 -- When building migrations remove models import when it's not needed

### PR DESCRIPTION
**Pull request by chrisjluc**
https://code.djangoproject.com/ticket/24514

### CODE REVIEW

File: tests/migrations/test_writer.py [hunk 1]
```diff
@@ -454,6 +454,22 @@ def test_sorted_imports(self):
             output
         )
 
+    def test_exclude_models_import(self):
+        """
+        #24514 - Test imports exclude django.db.models if it isn't needed.
+        """
+        migration = type(str("Migration"), (migrations.Migration,), {
+            "operations": [
```
Comment (MarkusH) [inline tests/migrations/test_writer.py]: I'd leave `'operations'` an empty list. Also, can you use single quotes here, please.

**Comment by MarkusH** [comment]
As already noted on the ticket, I think `models` should stay as a convenience import.

**Comment by MarkusH** [comment]
buildbot, test this please.

File: tests/migrations/test_writer.py (hunk 1 above)
Comment (MarkusH) [inline tests/migrations/test_writer.py]: Forget about the single quotes, we use them in this location all over this file.

**Comment by chrisjluc** [comment]
I think having `models` in a migration that doesn't use the import might spark confusion for why it's being imported in the first place. I think it makes the migrations cleaner by not having unused imports. I don't see many situations where someone would manually need the `models` import. But, in the rare occasion that they do need it, manually adding it is not that bad.

**Comment by timgraham** [comment]
merged in a7bc00e17bebe34cdf23f744937b3c386ee73d81, thanks!

**Pull request closed without merging**